manifests/fpm.pp

```diff
@@ -44,12 +44,12 @@
 #   Defaults is empty hash.
 #
 class php::fpm (
-  $ensure               = $::php::ensure,
+  $ensure               = $::php::params::ensure,
```

Comment:
Not 100% sure about this.  It's certainly not backwards compatible.
Currently you could do
```puppet
class {'php': ensure => absent}
```

I started looking at his the other day.  I think the problem is the way the class is tested.  It's supposed to be a private class only called from `init.pp`.  In the testing, it's tested as if it was independent.  You've probably noticed you can't just do a
```ruby
let(:pre_condition) { 'include ::php' }
```
without getting a duplicate resource declaration.  I thought about doing
```ruby
let(:pre_condition) { 'class { "::php": fpm => false' }
```
but that seems pretty hacky too.  Maybe the tests should just be moved into `php_spec.rb`?